src/esp/gfx/CubeMap.cpp

```diff
@@ -0,0 +1,396 @@
+                                             coordStrings[iFace]);
+      } break;
+
+      case TextureType::Depth: {
+        filename = Cr::Utility::formatString("{}.{}.{}.hdr", imageFilePrefix,
+                                             getTextureTypeFilenameString(type),
+                                             coordStrings[iFace]);
+      } break;
+    }
+    CORRADE_ASSERT(!filename.empty(),
+                   "CubeMap::saveTexture(): Unknown texture type.", false);
+
+    if (!converter->exportToFile(image, filename)) {
+      return false;
+    } else {
+      LOG(INFO) << "Saved image " << iFace << " to " << filename;
+    }
+  }
+
+  return true;
+}
+#endif
+
+void CubeMap::loadTexture(TextureType type,
+                          const std::string& imageFilePrefix,
+                          const std::string& imageFileExtension) {
+  // plugin manager used to instantiate importers which in turn are used
+  // to load image data
+  Cr::PluginManager::Manager<Mn::Trade::AbstractImporter> manager;
+  Cr::Containers::Pointer<Mn::Trade::AbstractImporter> importer =
+      manager.loadAndInstantiate("AnyImageImporter");
+  CORRADE_INTERNAL_ASSERT(importer);
+
+  const char* coordStrings[6] = {"+X", "-X", "+Y", "-Y", "+Z", "-Z"};
+  int imageSize = 0;
+
+  // set images
+  Mn::GL::CubeMapTexture* texture = nullptr;
+  switch (type) {
+    case TextureType::Color:
+      texture = textures_[TextureType::Color].get();
+      break;
+
+    case TextureType::Depth:
+      texture = textures_[TextureType::Depth].get();
+      break;
+  }
+  CORRADE_ASSERT(texture, "CubeMap::loadTexture(): Unknown texture type.", );
+
+  for (int iFace = 0; iFace < 6; ++iFace) {
+    // open image file
+
+    std::string filename = Cr::Utility::formatString(
+        "{}.{}.{}.{}", imageFilePrefix, getTextureTypeFilenameString(type),
+        coordStrings[iFace], imageFileExtension);
+
+    importer->openFile(filename);
+    Cr::Containers::Optional<Mn::Trade::ImageData2D> image =
+        importer->image2D(0);
+
+    // sanity checks
+    CORRADE_INTERNAL_ASSERT(image);
+    Mn::Vector2i size = image->size();
+    CORRADE_ASSERT(
+        size.x() == size.y(),
+        " CubeMap::loadTexture(): each texture image must be a square.", );
+    if (iFace == 0) {
+      imageSize = size.x();
+      reset(imageSize);
+    } else {
+      CORRADE_ASSERT(size.x() == imageSize,
+                     " CubeMap::loadTexture(): texture images must have the "
+                     "same size.", );
+    }
+
+    switch (type) {
+      case TextureType::Color:
+        texture->setSubImage(convertFaceIndexToCubeMapCoordinate(iFace), 0, {},
+                             *image);
+        break;
+
+      case TextureType::Depth: {
+        // R32F means 4 bytes per pixel
+        const int dim = static_cast<int>(std::sqrt(image->pixelSize() / 4));
+        // reinterpret the data as a R32F image
```

Comment:
@mosra: can I do it in this way?
What I am thinking is no matter what is the input's pixel format, whether it is RGB32F, or anything else, reinterpret it as R32F. 
Is this code correct?

Reply:
No, because the green channel of the first pixel will be now treated as the second pixel, the blue channel of the first pixel will be now treated as the third pixel and so on.

And the `dim` is also wrong. Completely. The `pixelSize()` is a *size of the pixel*, which was originally 4, now it will be 1. 

Did you even try to run the code? :)

Reply:
No, I did not have hdr images to try it. I should try it. My bad.
I simply do not have time to do it during the weekend, and I cannot do it in this PR, because it only introduces the CubeMap class. No one is calling it now.

Reply:
So can I just do:
```
        Mn::ImageView2D imageView(Mn::PixelFormat::R32F, image->size(),
                                  image->data());
                                  ```

Reply:
(Replied to the second comment on Slack.)

> I cannot do it in this PR, because it only introduces the CubeMap class.

Yes but what's the point of the PR having code that's just completely broken? Not put it at all there, then. In my opinion, a PR should ideally be *atomic*, that is a minimal but complete working piece of code, working correctly, documented and including tests. Anything less leads to technical debt and headaches.

Reply:
> Yes but what's the point of the PR having code that's just completely broken? 

I never say I will merge it like this. I am working on this code to get it correct. That is the purpose to have the code review, get advices from the reviewer, fix anything that needs to be fixed.

And when I say:
>I cannot do it in this PR, because it only introduces the CubeMap class.

Please do not get me wrong. I cannot test it here in this PR, it does not mean I would not test it. Actually I am testing it in my other branch called "cubemap".

>In my opinion, a PR should ideally be atomic, that is a minimal but complete working piece of code, working correctly, documented and including tests. 

I agree. We are on the same page. Again, I never, ever meant to push it in current shape. I am still working on it.